Add DPI awareness

carldebilly opened the pull request:
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno.hotdesign/issues/4139

# Bugfix

## What is the new behavior?

- Introduced `SetProcessDpiAwareness()` to address DPI scaling issues when not defined in the process manifest (when started from `dotnet run`)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [X] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

carldebilly committed:
fix(win32_dpi): Add DPI awareness support and improve XAML method naming

Introduced SetProcessDpiAwareness() to address DPI scaling issues when running from dotnet CLI.

carldebilly commented:
@MartinZikmund @dr1rrb Please review this. It solve the problem when the process is started using `dotnet run` because there's no RT_MANIFEST in this executable declaring a process-wide dpi awareness.

This change MAY break uno island, so maybe there's a better approach to it. Since the `SetThreadDpiAwarenessContext()` seems to not apply on pointers (they are coming from a different thread?)

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-20443/wasm-skia-net9/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-20443/docs/index.html

nventive-devops commented:
The [build 165973](https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=165973) found UI Test snapshots differences: `android-28-net8`: **20**, `android-28-net8-Snap`: **41**, `ios`: **4**, `ios-Snap`: **35**, `skia-linux-screenshots`: **69**, `skia-windows-screenshots`: **168**, `wasm`: **47**, `wasm-automated-net9.0-WinUI-Benchmarks-automated`: **0**, `wasm-automated-net9.0-WinUI-Default-automated`: **15**, `wasm-automated-net9.0-WinUI-RuntimeTests-0`: **0**, `wasm-automated-net9.0-WinUI-RuntimeTests-1`: **0**, `wasm-automated-net9.0-WinUI-RuntimeTests-2`: **0**

<details>
<summary>Details</summary>


* `android-28-net8`: **20** changed over 824

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `Detereminate_ProgressRing_Validation50_[#FF0000_#008000_#008000_#FF0000]_Progress-Ring-Value-50`
  - `FlyoutTest_When_NoOverlayInputPassThroughElement_Then_DontPassThrough_woOff_UITests_Shared_Windows_UI_Xaml_Controls_Flyout_Flyout_OverlayInputPassThroughElement`
  - `Detereminate_ProgressRing_Validation25_[#FF0000_#008000_#FF0000_#FF0000]_Progress-Ring-Value-25`
  - `Detereminate_ProgressRing_Validation75_[#FF0000_#008000_#008000_#008000]_Progress-Ring-Value-75`
  - `ListView_SelectedItems_SamplesApp_Windows_UI_Xaml_Controls_ListView_ListViewSelectedItems`
  - `ProgressRing_IsEnabled_Running_UITests_Windows_UI_Xaml_Controls_ProgressRing_WindowsProgressRing_GH1220`
  - `SequentialAnimations_SamplesApp_Windows_UI_Xaml_Media_Animation_SequentialAnimationsPage`
  - `When_Parent_PointerMoved_After_drag_on_ScrollViewer_-_touch`
  - `When_SingleSelectionWithItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `When_SingleSelectionWithoutItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `When_BitmapIcon_Generic_Changed`
  - `When_BitmapIcon_Generic_Initial`
  - `When_BitmapIcon_Generic_UITests_Shared_Windows_UI_Xaml_Controls_BitmapIconTests_BitmapIcon_Generic`
  - `UpDownEnabledTest_UITests_Shared_Microsoft_UI_Xaml_Controls_NumberBoxTests_NumberBoxPage`
  - `ProgressRing_Visibility_Collapsed_UITests_Windows_UI_Xaml_Controls_ProgressRing_WindowsProgressRing_GH1220`
  - `When_ExtendedSelectionWithoutItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `When_NoSelectionWithItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `When_Parent_PointerMoved_After_drag_on_non-scrolling_ScrollViewer`
  - `Button_IsOpacity_Validation_Uno_UI_Samples_Content_UITests_ButtonTestsControl_Button_Opacity_Automated`
  - `DecimalFormatterTest_UITests_Shared_Microsoft_UI_Xaml_Controls_NumberBoxTests_NumberBoxPage`
  </details>

* `android-28-net8-Snap`: **41** changed over 1067

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `MediaPlayerElement_Sources_UITests_Shared_Windows_UI_Xaml_Controls_MediaPlayerElement_MediaPlayerElement_Sources`
  - `MediaPlayerElement_Using_3gp_UITests_Shared_Windows_UI_Xaml_Controls_MediaPlayerElement_MediaPlayerElement_3gp_Extension`
  - `MediaPlayerElement_Using_mp3_Audio_only_UITests_Shared_Windows_UI_Xaml_Controls_MediaPlayerElement_MediaPlayerElement_Mp3_Extension`
  - `MediaPlayerElement_Using_ogg_UITests_Shared_Windows_UI_Xaml_Controls_MediaPlayerElement_MediaPlayerElement_Ogg_Extension`
  - `SwipeControl_MUXControlsTestApp_SwipeControlPage2_MUXControlsTestApp_SwipeControlPage2`
  - `Gesture_Recognizer_Pointer_Events_test_bench_UITests_Shared_Windows_UI_Input_GestureRecognizer_PointersEvents`
  - `Buttons_UITests_Microsoft_UI_Xaml_Controls_RadioButtonsTests_RadioButtonsPage_UITests_Microsoft_UI_Xaml_Controls_RadioButtonsTests_RadioButtonsPage`
  - `MediaPlayerElement_Mini_player_UITests_Shared_Windows_UI_Xaml_Controls_MediaPlayerElement_MediaPlayerElement_Minimal`
  - `Scrolling_MUXControlsTestApp_ScrollViewDynamicPage_MUXControlsTestApp_ScrollViewDynamicPage`
  - `Scrolling_MUXControlsTestApp_ScrollViewKeyboardAndGamepadNavigationPage_MUXControlsTestApp_ScrollViewKeyboardAndGamepadNavigationPage`
  - `Image_UITests_Windows_UI_Xaml_Controls_ImageTests_SvgImageSource_Icons_UITests_Windows_UI_Xaml_Controls_ImageTests_SvgImageSource_Icons`
  - `MUX_NumberBox_UITests_Shared_Microsoft_UI_Xaml_Controls_NumberBoxTests_NumberBoxPage`
  - `RatingControl_UITests_Microsoft_UI_Xaml_Controls_RatingControlTests_RatingControlPage_UITests_Microsoft_UI_Xaml_Controls_RatingControlTests_RatingControlPage`
  - `NavigationView_MUXControlsTestApp_NavigationViewCustomThemeResourcesPage_MUXControlsTestApp_NavigationViewCustomThemeResourcesPage`
  - `TeachingTip_UITests_Microsoft_UI_Xaml_Controls_TeachingTipTests_TeachingTipPage_UITests_Microsoft_UI_Xaml_Controls_TeachingTipTests_TeachingTipPage`
  - `TextBlock_UITests_Shared_Windows_UI_Xaml_Controls_TextBlockControl_TextBlock_Layout_UITests_Shared_Windows_UI_Xaml_Controls_TextBlockControl_TextBlock_Layout`
  - `ListView_ListViewSelectedItems_SamplesApp_Windows_UI_Xaml_Controls_ListView_ListViewSelectedItems`
  - `NavigationView_MUXControlsTestApp_NavigationViewRS4Page_MUXControlsTestApp_NavigationViewRS4Page`
  - `NavigationView_MUXControlsTestApp_NavigationViewTopNavOnlyPage_MUXControlsTestApp_NavigationViewTopNavOnlyPage`
  - `NavigationView_MUXControlsTestApp_NavigationViewTopNavPage_MUXControlsTestApp_NavigationViewTopNavPage`
  </details>

* `ios`: **4** changed over 255

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `Check_ListView_Swallows_Measure_UITests_Shared_Windows_UI_Xaml_Controls_ListView_ListView_With_ListViews_Count_Measure`
  - `TextBox_UpdatedBinding_On_OneWay_Mode_UITests_Windows_UI_Xaml_Controls_TextBox_TextBox_Bindings`
  - `ProgressRing_Visibility_Collapsed_UITests_Windows_UI_Xaml_Controls_ProgressRing_WindowsProgressRing_GH1220`
  - `ProgressRing_IsEnabled_Running_UITests_Windows_UI_Xaml_Controls_ProgressRing_WindowsProgressRing_GH1220`
  </details>

* `ios-Snap`: **35** changed over 984

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `Icons_UITests_Shared_Windows_UI_Xaml_Controls_BitmapIconTests_BitmapIcon_Foreground_UITests_Shared_Windows_UI_Xaml_Controls_BitmapIconTests_BitmapIcon_Foreground`
  - `Brushes_EllipsemaskingEllipseGrid_Uno_UI_Samples_UITests_ImageBrushTestControl_EllipsemaskingEllipseGrid`
  - `Brushes_BorderImageBrush_Uno_UI_Samples_UITests_ImageBrushTestControl_BorderImageBrush`
  - `Brushes_ImageBrushWithScaleTransform_Uno_UI_Samples_UITests_ImageBrushTestControl_ImageBrushWithScaleTransform`
  - `Focus_UITests_Windows_UI_Xaml_FocusTests_Focus_FocusCycle_UITests_Windows_UI_Xaml_FocusTests_Focus_FocusCycle`
  - `Image_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Wider_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Wider`
  - `MediaPlayerElement_Sources_UITests_Shared_Windows_UI_Xaml_Controls_MediaPlayerElement_MediaPlayerElement_Sources`
  - `SwipeControl_MUXControlsTestApp_SwipeControlPage2_MUXControlsTestApp_SwipeControlPage2`
  - `Brushes_ImageBrushWithCompositeTransform_Uno_UI_Samples_UITests_ImageBrushTestControl_ImageBrushWithCompositeTransform`
  - `Brushes_RectangleStretchFill_Uno_UI_Samples_UITests_ImageBrushTestControl_RectangleStretchFill`
  - `Scrolling_MUXControlsTestApp_ScrollViewDynamicPage_MUXControlsTestApp_ScrollViewDynamicPage`
  - `UIElement_TransformToVisual_Simple_UITests_Shared_Windows_UI_Xaml_UIElementTests_TransformToVisual_Simple`
  - `Default_SamplesApp_Wasm_Windows_UI_Xaml_Controls_ListView_ListView_IsSelected_SamplesApp_Wasm_Windows_UI_Xaml_Controls_ListView_ListView_IsSelected`
  - `Image_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Bigger_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Bigger`
  - `Image_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Equal_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Equal`
  - `Image_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Smaller_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Smaller`
  - `Image_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Taller_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Taller`
  - `WebView_UITests_Microsoft_UI_Xaml_Controls_WebView2Tests_WebView2_NavigationProperties_UITests_Microsoft_UI_Xaml_Controls_WebView2Tests_WebView2_NavigationProperties`
  - `GridView_GenericApp_Views_Samples_Shared_Content_UITests_GridViewGrouped_GenericApp_Views_Samples_Shared_Content_UITests_GridViewGrouped`
  - `Image_Uno_UI_Samples_UITests_Image_Image_Stretch_Uno_UI_Samples_UITests_Image_Image_Stretch`
  </details>

* `skia-linux-screenshots`: **69** changed over 2246

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `CalendarView_Theming.png-dark`
  - `DisplayInformation.png-dark`
  - `DisplayInformation.png`
  - `ImageBrushAlignmentXY.png-dark`
  - `ImageBrushAlignmentXY.png`
  - `ButtonClippingTestsControl.png-dark`
  - `ClipboardTests.png-dark`
  - `ClipboardTests.png`
  - `Examples.png`
  - `ExpanderColorValidationPage.png-dark`
  - `ExpanderColorValidationPage.png`
  - `Ellipse.png-dark`
  - `Ellipse.png`
  - `DragDrop_TestPage.png-dark`
  - `DragDrop_TestPage.png`
  - `ImageBrushInList.png-dark`
  - `ImageBrushInList.png`
  - `Gamepad_Enumeration.png-dark`
  - `Gamepad_Enumeration.png`
  - `Buttons.png-dark`
  </details>

* `skia-windows-screenshots`: **168** changed over 2246

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `ApplicationViewSizing.png`
  - `BenchmarkDotNet.png-dark`
  - `CommandBar_BackButtonTitle.png-dark`
  - `CommandBar_BackButtonTitle.png`
  - `CommandBar_Background.png-dark`
  - `CommandBar_Background.png`
  - `AnimatedIconPage.png`
  - `AppWindowPresenters.png`
  - `ClipboardTests.png`
  - `Focus_FocusVisual_Properties.png-dark`
  - `Focus_FocusVisual_Properties.png`
  - `AppWindowClosing.png`
  - `AppWindowClosing.png-dark`
  - `AppWindowPresenters.png-dark`
  - `BasicEnability.png-dark`
  - `Buttons.png`
  - `ButtonClippingTestsControl.png-dark`
  - `ComboBox_ItemDataContext.png-dark`
  - `ComboBox_ItemDataContext.png`
  - `Examples.png`
  </details>

* `wasm`: **47** changed over 1051

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `GenericApp.Views.Samples.Shared.Content.UITests.GridViewMultipleSelectionMode`
  - `SamplesApp.Windows_UI_Xaml_Controls.ListView.ListViewSelectedItems`
  - `SamplesApp.Windows_UI_Xaml_Controls.ListView.ListViewHeaderFooter_ImplicitlyBinded`
  - `UITests.Shared.Windows_UI_Xaml_Controls.MediaPlayerElement.MediaPlayerElement_Minimal`
  - `UITests.Shared.Windows_UI_Xaml_Controls.MediaPlayerElement.MediaPlayerElement_Ogg_Extension`
  - `UITests.Windows_UI_Xaml_Controls.CommandBar.BackButtonTitle.CommandBar_Frame`
  - `UITests.Windows_UI_Xaml_Controls.ImageTests.SvgImageSource_FromMsAppData`
  - `UITests.Windows_UI_Xaml_Controls.ImageTests.SvgImageSource_FromStream`
  - `Uno.UI.Samples.Content.UITests.ButtonTestsControl.RadioButton_Multiple_Unnamed_Groups`
  - `SamplesApp.Wasm.Windows_UI_Xaml_Controls.ComboBox.ComboBox_Corners`
  - `SamplesApp.Windows_UI_Xaml_Controls.ListView.ListView_Expandable_Item_Large`
  - `UITests.Windows_UI_Xaml_Controls.ListView.ListViewItem_IsEnabled`
  - `UITests.Windows_UI_Xaml_Controls.ListView.ListView_DuplicateItem`
  - `UITests.Windows_UI_Xaml_Media.AcrylicBrushTests.BasicAcrylicBrushTest`
  - `Uno.UI.Samples.Content.UITests.ButtonTestsControl.CheckBox_Button`
  - `UITests.Shared.Windows_UI_Input.PointersTests.EventsSequences`
  - `UITests.Windows_UI_Xaml.FocusTests.Focus_FocusState`
  - `UITests.Windows_UI_Xaml_Controls.Repeater.ItemsRepeater_Nested`
  - `Uno.UI.Samples.Content.UITests.GridView.GridViewVariableItemHeight`
  - `UITests.Shared.Windows_UI_Xaml_Controls.Button.Overlapped_Buttons`
  </details>

* `wasm-automated-net9.0-WinUI-Benchmarks-automated`: **0** changed over 1

* `wasm-automated-net9.0-WinUI-Default-automated`: **15** changed over 876

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `Default_StrokeThickness_MyPolygon`
  - `SequentialAnimations_SamplesApp_Windows_UI_Xaml_Media_Animation_SequentialAnimationsPage`
  - `When_Padding_Is_Changed_Then_Cache_Is_Missed_UITests_Windows_UI_Xaml_Controls_TextBlockControl_TextBlock_MeasureCache`
  - `When_SingleSelectionWithItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `When_SingleSelectionWithoutItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `Default_StrokeThickness_MyPath`
  - `When_MaxLines_Then_AlignmentPositionIsCorrect_Uno_UI_Samples_Content_UITests_TextBlockControl_SimpleText_MaxLines_Two_With_Wrap`
  - `Default_StrokeThickness_MyLine`
  - `ListView_SelectedItems_SamplesApp_Windows_UI_Xaml_Controls_ListView_ListViewSelectedItems`
  - `Default_StrokeThickness_MyPolyline`
  - `TestProgressRing_InitialState_UITests_Microsoft_UI_Xaml_Controls_ProgressRing_WinUIProgressRing_Features`
  - `Default_StrokeThickness_MyEllipse`
  - `When_TextTrimming_Is_Set_Then_Ellipsis_Is_Used_UITests_Windows_UI_Xaml_Controls_TextBlockControl_TextBlock_TextTrimming`
  - `When_NoSelection_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `When_TransformToVisual_ScrollViewer_UITests_Shared_Windows_UI_Xaml_UIElementTests_TransformToVisual_ScrollViewer`
  </details>

* `wasm-automated-net9.0-WinUI-RuntimeTests-0`: **0** changed over 1

* `wasm-automated-net9.0-WinUI-RuntimeTests-1`: **0** changed over 1

* `wasm-automated-net9.0-WinUI-RuntimeTests-2`: **0** changed over 1

</details>

MartinZikmund commented:
Seems good, but will also let @ramezgerges chime in 👍

ramezgerges commented on src/Uno.UI.Runtime.Skia.Win32/Win32Host.cs:
```diff
@@ -129,6 +132,42 @@ protected override void Initialize()
+		//   </asmv3:application>
+		// </assembly>
+		//
+		// Important: Calling SetThreadDpiAwarenessContext() affects only the DPI behavior of user32/GDI calls **on that thread**,
+		// but it does NOT update the global DPI-awareness context of the process, nor does it affect how the OS scales
+		// the window or reports DPI-related metrics to frameworks like Skia or WinUI. To enable proper DPI scaling and
+		// accurate coordinate transformation, the process DPI-awareness must be set at launch or via SetProcessDpiAwarenessContext().
```
That's not accurate. DPI scaling was already working accurately (i.e. actual scaling not just bitmap stretching), at least when running the executable directly. Also, Skia doesn't read DPI-related metrics, so I'm not sure what that's about.

ebariche commented on src/Uno.UI.Runtime.Skia.Win32/Win32Host.cs:
```diff
@@ -129,6 +132,42 @@ protected override void Initialize()
 
+	private void SetProcessDpiAwareness()
+	{
+		// This call ensures that the application is properly marked as per-monitor DPI aware (PerMonitorV2).
+		// When launched via `dotnet.exe` (e.g., `dotnet MyApp.dll` or `dotnet run`), the host process is the .NET CLI runtime,
+		// which does NOT contain a DPI-awareness declaration in its embedded application manifest (RT_MANIFEST).
+		// Consequently, Windows assumes the process is DPI-unaware, applies bitmap scaling, and reports a logical DPI of 96,
+		// leading to incorrect pointer coordinates, layout glitches, and drag-and-drop offsets.
+		//
+		// In contrast, when launching a native application executable (e.g., `MyApp.exe`), the process inherits the
+		// DPI-awareness setting defined in the manifest block:
```
@ramezgerges : The presence of the bug indicates that setting it on the main thread is not sufficient. Setting it at the thread level is meant to override the process-wide default which is either set in the manifest or by `SetProcessDpiAwarenessContext`. See : https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setprocessdpiawarenesscontext#remarks

carldebilly committed:
chore: Refactor DPI awareness initialization logic

Move DPI awareness setup to a dedicated `DpiBootstrap` class using a module initializer for improved clarity and reuse. This ensures the correct DPI context is applied before any window initialization, replacing redundant logic in `Win32Host`.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-20443/wasm-skia-net9/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-20443/docs/index.html

unodevops commented:
⚠️⚠️ The build [166322](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=166322) has failed on Uno.UI - CI.

jeromelaban commented:
@mergifyio backport release/stable/6.0

nventive-devops commented:
The [build 166322](https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=166322) found UI Test snapshots differences: `android-28-net8`: **26**, `android-28-net8-Snap`: **51**, `ios`: **4**, `ios-Snap`: **49**, `skia-linux-screenshots`: **61**, `skia-windows-screenshots`: **61**, `wasm`: **54**, `wasm-automated-net9.0-WinUI-Benchmarks-automated`: **0**, `wasm-automated-net9.0-WinUI-Default-automated`: **14**, `wasm-automated-net9.0-WinUI-RuntimeTests-0`: **0**, `wasm-automated-net9.0-WinUI-RuntimeTests-1`: **1**, `wasm-automated-net9.0-WinUI-RuntimeTests-2`: **1**

<details>
<summary>Details</summary>


* `android-28-net8`: **26** changed over 824

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `Detereminate_ProgressRing_Validation25_[#FF0000_#008000_#FF0000_#FF0000]_Progress-Ring-Value-25`
  - `Detereminate_ProgressRing_Validation50_[#FF0000_#008000_#008000_#FF0000]_Progress-Ring-Value-50`
  - `DoesNotInterfereWithHorizontalDrag_UITests_Microsoft_UI_Xaml_Controls_RefreshContainerTests_RefreshContainerHorizontalScroll`
  - `FocusManager_GetFocusedElement_Button_Validation_FocusManager_-_GetFocusedElement_-_Button_-_2_-_After_Selection`
  - `FocusManager_GetFocusedElement_Button_Validation_FocusManager_-_GetFocusedElement_-_Button_-_1_-_Initial_State`
  - `FocusManager_FocusDirection_Down_Validation_Uno_UI_Samples_Content_UITests_FocusTests_FocusManager_FocusDirection`
  - `FocusManager_GetFocusedElement_Button_Validation_Uno_UI_Samples_Content_UITests_FocusTests_FocusManager_GetFocus_Automated`
  - `Detereminate_ProgressRing_Validation0_[#FF0000_#FF0000_#FF0000_#FF0000]_UITests_Microsoft_UI_Xaml_Controls_ProgressRing_WinUIDeterminateProgressRing`
  - `NativeCommandBar_Size_Uno_UI_Samples_Content_UITests_CommandBar_CommandBar_Dynamic`
  - `ProgressRing_IsEnabled_Running_UITests_Windows_UI_Xaml_Controls_ProgressRing_WindowsProgressRing_GH1220`
  - `ProgressRing_Visibility_Collapsed_UITests_Windows_UI_Xaml_Controls_ProgressRing_WindowsProgressRing_GH1220`
  - `When_Clipped_Rounded_Corners_UITests_Windows_UI_Xaml_Clipping_Clipping4273`
  - `When_Clip_Is_Set_On_Child_Element_original`
  - `When_MultipleSelectionWithoutItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `UpDownEnabledTest_UITests_Shared_Microsoft_UI_Xaml_Controls_NumberBoxTests_NumberBoxPage`
  - `When_ExtendedSelectionWithoutItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `When_Clip_Is_Set_On_Child_Element_validation`
  - `When_Gyrometer_Is_Retrived_With_GetDefault_UITests_Shared_Windows_Devices_GyrometerTests`
  - `When_SingleSelectionWithItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `When_Click_Outside_Dialog_Expect_No_Dismiss_UITests_Shared_MessageDialogTests_MessageDialogTest`
  </details>

* `android-28-net8-Snap`: **51** changed over 1067

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `Gesture_Recognizer_Pointer_Events_test_bench_UITests_Shared_Windows_UI_Input_GestureRecognizer_PointersEvents`
  - `NavigationView_MUXControlsTestApp_NavigationViewCustomThemeResourcesPage_MUXControlsTestApp_NavigationViewCustomThemeResourcesPage`
  - `SwipeControl_MUXControlsTestApp_SwipeControlPage2_MUXControlsTestApp_SwipeControlPage2`
  - `NavigationView_MUXControlsTestApp_NavigationViewRS4Page_MUXControlsTestApp_NavigationViewRS4Page`
  - `ListView_ListViewSelectedItems_SamplesApp_Windows_UI_Xaml_Controls_ListView_ListViewSelectedItems`
  - `MediaPlayerElement_Using_3gp_UITests_Shared_Windows_UI_Xaml_Controls_MediaPlayerElement_MediaPlayerElement_3gp_Extension`
  - `MediaPlayerElement_Using_mp3_Audio_only_UITests_Shared_Windows_UI_Xaml_Controls_MediaPlayerElement_MediaPlayerElement_Mp3_Extension`
  - `MediaPlayerElement_Using_ogg_UITests_Shared_Windows_UI_Xaml_Controls_MediaPlayerElement_MediaPlayerElement_Ogg_Extension`
  - `ListView_ListView_FirstLastCacheIndex_SamplesApp_Windows_UI_Xaml_Controls_ListView_ListView_FirstLastCacheIndex`
  - `RatingControl_UITests_Microsoft_UI_Xaml_Controls_RatingControlTests_RatingControlPage_UITests_Microsoft_UI_Xaml_Controls_RatingControlTests_RatingControlPage`
  - `MediaPlayerElement_Mini_player_UITests_Shared_Windows_UI_Xaml_Controls_MediaPlayerElement_MediaPlayerElement_Minimal`
  - `Image_UITests_Windows_UI_Xaml_Controls_ImageTests_SvgImageSource_Icons_UITests_Windows_UI_Xaml_Controls_ImageTests_SvgImageSource_Icons`
  - `Image_UITests_Windows_UI_Xaml_Controls_ImageTests_SvgImageSource_NoSize_UITests_Windows_UI_Xaml_Controls_ImageTests_SvgImageSource_NoSize`
  - `MUX_NumberBox_UITests_Shared_Microsoft_UI_Xaml_Controls_NumberBoxTests_NumberBoxPage`
  - `WebView_UITests_Windows_UI_Xaml_Controls_WebView_WebView_Title_UITests_Windows_UI_Xaml_Controls_WebView_WebView_Title`
  - `UIElement_TransformToVisual_Simple_UITests_Shared_Windows_UI_Xaml_UIElementTests_TransformToVisual_Simple`
  - `UIElement_TransformToVisual_Transform_UITests_Shared_Windows_UI_Xaml_UIElementTests_TransformToVisual_Transform`
  - `UIElement_UITests_Shared_Windows_UI_Xaml_UIElementTests_UIElement_Layout_Constrains_UITests_Shared_Windows_UI_Xaml_UIElementTests_UIElement_Layout_Constrains`
  - `UIElement_UITests_Shared_Windows_UI_Xaml_UIElementTests_UIElement_Native_Child_UITests_Shared_Windows_UI_Xaml_UIElementTests_UIElement_Native_Child`
  - `TeachingTip_UITests_Microsoft_UI_Xaml_Controls_TeachingTipTests_TeachingTipPage_UITests_Microsoft_UI_Xaml_Controls_TeachingTipTests_TeachingTipPage`
  </details>

* `ios`: **4** changed over 255

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `When_SingleSelectionWithoutItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `ProgressRing_Visibility_Collapsed_UITests_Windows_UI_Xaml_Controls_ProgressRing_WindowsProgressRing_GH1220`
  - `TextBox_UpdatedBinding_On_OneWay_Mode_UITests_Windows_UI_Xaml_Controls_TextBox_TextBox_Bindings`
  - `ContentControl_Changing_ContentTemplate_Uno_UI_Samples_Content_UITests_ContentControlTestsControl_ContentControl_Changing_ContentTemplate`
  </details>

* `ios-Snap`: **49** changed over 984

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `Brushes_ImageBrushWithRotateTransform_Uno_UI_Samples_UITests_ImageBrushTestControl_ImageBrushWithRotateTransform`
  - `Default_Uno_UI_Samples_Content_UITests_WebView_WebView_AnchorNavigation_Uno_UI_Samples_Content_UITests_WebView_WebView_AnchorNavigation`
  - `Brushes_ImageBrushStretch2_Uno_UI_Samples_UITests_ImageBrushTestControl_ImageBrushStretch2`
  - `Brushes_UITests_Windows_UI_Xaml_Media_BrushesTests_RevealBrush_Fallback_UITests_Windows_UI_Xaml_Media_BrushesTests_RevealBrush_Fallback`
  - `Brushes_Uno_UI_Samples_Samples_Shared_Content_UITests_ImageBrushInList_Uno_UI_Samples_Samples_Shared_Content_UITests_ImageBrushInList`
  - `ColorPicker_WinUIColorPickerPage_UITests_Shared_Microsoft_UI_Xaml_Controls_ColorPickerTests_WinUIColorPickerPage`
  - `ColorPicker_ColorPickerSample_UITests_Microsoft_UI_Xaml_Controls_ColorPickerTests_ColorPickerSample`
  - `Image_UITests_Shared_Windows_UI_Xaml_Controls_ImageTests_Image_Stretch_Algmnt_Inf_Vertical_UITests_Shared_Windows_UI_Xaml_Controls_ImageTests_Image_Stretch_Algmnt_Inf_Vertical`
  - `Image_UITests_Shared_Windows_UI_Xaml_Controls_ImageTests_Image_Stretch_Alignment_SizeOnControl_UITests_Shared_Windows_UI_Xaml_Controls_ImageTests_Image_Stretch_Alignment_SizeOnControl`
  - `NavigationView_FluentStyle_NavigationViewSample_SamplesApp_Samples_Microsoft_UI_Xaml_Controls_NavigationViewTests_FluentStyle_FluentStyle_NavigationViewSample`
  - `TextBlock_TextBoxSizeChanging_Uno_UI_Samples_Content_UITests_TextBlockControl_TextBoxSizeChanging`
  - `NavigationView_MUXControlsTestApp_NavigationViewMenuItemStretchPage_MUXControlsTestApp_NavigationViewMenuItemStretchPage`
  - `NavigationView_MUXControlsTestApp_NavigationViewMinimalPage_MUXControlsTestApp_NavigationViewMinimalPage`
  - `TextBlock_TextBlockMultilineInStarStackPanel_Uno_UI_Samples_Content_UITests_TextBlockControl_TextBlockMultilineInStarStackPanel`
  - `WebView_UITests_Microsoft_UI_Xaml_Controls_WebView2Tests_WebView2_ExecuteScriptAsync_UITests_Microsoft_UI_Xaml_Controls_WebView2Tests_WebView2_ExecuteScriptAsync`
  - `WebView_UITests_Microsoft_UI_Xaml_Controls_WebView2Tests_WebView2_NavigationProperties_UITests_Microsoft_UI_Xaml_Controls_WebView2Tests_WebView2_NavigationProperties`
  - `Brushes_BorderImageBrush_Uno_UI_Samples_UITests_ImageBrushTestControl_BorderImageBrush`
  - `WebView_WebView2_Javascript_AlertConfirmPrompt_SamplesApp_Microsoft_UI_Xaml_Controls_WebView2Tests_WebView2ControlJavaScriptAlertConfirmPrompt`
  - `TextBlock_TextBlock_FixedWidth_With_DataBound_Run_Uno_UI_Samples_Content_UITests_TextBlockControl_TextBlock_FixedWidth_With_DataBound_Run`
  - `Brushes_PanelImageBrush_Uno_UI_Samples_UITests_ImageBrushTestControl_PanelImageBrush`
  </details>

* `skia-linux-screenshots`: **61** changed over 2246

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `Gamepad_Enumeration.png-dark`
  - `Gamepad_Enumeration.png`
  - `ButtonClippingTestsControl.png-dark`
  - `DropDownButtonPage.png-dark`
  - `DropDownButtonPage.png`
  - `ClipboardTests.png-dark`
  - `ClipboardTests.png`
  - `Examples.png`
  - `ExpanderColorValidationPage.png-dark`
  - `ExpanderColorValidationPage.png`
  - `Gamepad_CurrentReading.png-dark`
  - `Gamepad_CurrentReading.png`
  - `Image_Stretch_None.png-dark`
  - `Image_Stretch_None.png`
  - `ImageIconPage.png-dark`
  - `ImageIconPage.png`
  - `DisplayInformation.png-dark`
  - `DisplayInformation.png`
  - `LargeImageList.png-dark`
  - `LargeImageList.png`
  </details>

* `skia-windows-screenshots`: **61** changed over 2246

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `ButtonClippingTestsControl.png-dark`
  - `DropDownButtonPage.png-dark`
  - `DropDownButtonPage.png`
  - `Focus_WASM_Tab.png-dark`
  - `Focus_WASM_Tab.png`
  - `DragDrop_ListView_Custom_States.png-dark`
  - `DragDrop_ListView_Custom_States.png`
  - `DragDrop_TestPage.png-dark`
  - `DragDrop_TestPage.png`
  - `ImageBrushInList.png-dark`
  - `ImageBrushInList.png`
  - `FocusManager_FocusDirection.png-dark`
  - `FocusManager_FocusDirection.png`
  - `Buttons.png`
  - `CalendarView_Theming.png`
  - `ClipboardTests.png-dark`
  - `DisplayInformation.png-dark`
  - `DisplayInformation.png`
  - `LargeImageList.png-dark`
  - `LargeImageList.png`
  </details>

* `wasm`: **54** changed over 1051

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `SamplesApp.Windows_UI_Xaml_Controls.ListView.ListViewSelectedItems`
  - `UITests.Shared.Windows_UI_Xaml_Controls.ComboBox.ComboBox_DropDownPlacement`
  - `UITests.Windows_UI_Xaml_Media_Animation.DoubleAnimation_FinalState_Opacity`
  - `UITests.Windows_UI_Xaml_Controls.ListView.ListViewItem_IsEnabled`
  - `UITests.Windows_UI_Xaml_Controls.ListView.ListView_DisplayMemberPath`
  - `UITests.Windows_UI_Xaml_Controls.ListView.ListView_DuplicateItem`
  - `UITests.Windows_UI_Xaml_Controls.SwipeControlTests.SwipeControl_ListView_ItemClick`
  - `Uno.UI.Samples.Content.UITests.CommandBar.CommandBar_Padding`
  - `UITests.Microsoft_UI_Xaml_Controls.InfoBadgeTests.InfoBadgePage`
  - `UITests.Shared.Windows_UI_Xaml_Controls.ContentDialogTests.ContentDialog_Closing`
  - `UITests.Shared.Windows_UI_Xaml_Controls.ContentDialogTests.ContentDialog_Simple`
  - `UITests.Windows_UI_Xaml.DragAndDrop.DragDrop_ListView_Custom_States`
  - `SamplesApp.Windows_UI_Xaml_Controls.ListView.ListViewHeaderFooter_ImplicitlyBinded`
  - `SamplesApp.Windows_UI_Xaml_Controls.ListView.UndefinedHeightListView`
  - `UITests.Microsoft_UI_Xaml_Controls.WebView2Tests.WebView2_NavigationProperties`
  - `UITests.Shared.Windows_UI_Xaml_Controls.ComboBox.ComboBox_VisibleBounds`
  - `UITests.Shared.Windows_UI_Xaml_Controls.CommandBar.CommandBar_Xaml_Automated`
  - `UITests.Windows_UI_Xaml_Controls.BorderTests.BorderWithNullBrushAndNonZeroThickness`
  - `Uno.UI.Samples.Content.UITests.ButtonTestsControl.AppBar_KeyBoard`
  - `Uno.UI.Samples.Content.UITests.ButtonTestsControl.Buttons`
  </details>

* `wasm-automated-net9.0-WinUI-Benchmarks-automated`: **0** changed over 1

* `wasm-automated-net9.0-WinUI-Default-automated`: **14** changed over 876

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `When_MaxLines_Then_AlignmentPositionIsCorrect_Uno_UI_Samples_Content_UITests_TextBlockControl_SimpleText_MaxLines_Two_With_Wrap`
  - `Default_StrokeThickness_MyRect`
  - `ListView_SelectedItems_SamplesApp_Windows_UI_Xaml_Controls_ListView_ListViewSelectedItems`
  - `Default_StrokeThickness_MyPolyline`
  - `Default_StrokeThickness_MyPath`
  - `Default_StrokeThickness_MyLine`
  - `SequentialAnimations_SamplesApp_Windows_UI_Xaml_Media_Animation_SequentialAnimationsPage`
  - `When_Padding_Is_Changed_Then_Cache_Is_Missed_UITests_Windows_UI_Xaml_Controls_TextBlockControl_TextBlock_MeasureCache`
  - `When_TextTrimming_Is_Set_Then_Ellipsis_Is_Used_UITests_Windows_UI_Xaml_Controls_TextBlockControl_TextBlock_TextTrimming`
  - `TestProgressRing_InitialState_UITests_Microsoft_UI_Xaml_Controls_ProgressRing_WinUIProgressRing_Features`
  - `When_SingleSelectionWithItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `When_NoSelection_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `Default_StrokeThickness_MyEllipse`
  - `Default_StrokeThickness_MyPolygon`
  </details>

* `wasm-automated-net9.0-WinUI-RuntimeTests-0`: **0** changed over 1

* `wasm-automated-net9.0-WinUI-RuntimeTests-1`: **1** changed over 1

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `RunRuntimeTests_SamplesApp_Samples_UnitTests_UnitTestsPage`
  </details>

* `wasm-automated-net9.0-WinUI-RuntimeTests-2`: **1** changed over 1

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `RunRuntimeTests_SamplesApp_Samples_UnitTests_UnitTestsPage`
  </details>

</details>
